openpype/hosts/maya/plugins/publish/validate_resolution.py

```diff
@@ -0,0 +1,100 @@
+import pyblish.api
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin
+)
+from maya import cmds
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.maya.api import lib
+from openpype.hosts.maya.api.lib import reset_scene_resolution
+
+
+class ValidateResolution(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+    """Validate the render resolution setting aligned with DB"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    families = ["renderlayer"]
+    hosts = ["maya"]
+    label = "Validate Resolution"
+    actions = [RepairAction]
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        invalid = self.get_invalid_resolution(instance)
+        if invalid:
+            raise PublishValidationError(
+                "issues occurred", description=(
+                    "Wrong render resolution setting. "
+                    "Please use repair button to fix it.\n"
+                    "If current renderer is V-Ray, "
+                    "make sure vraySettings node has been created"))
```

Comment:
Haha, I know this was my pseudocode example - but we can be slightly more descriptive in the message itself.
```suggestion
            raise PublishValidationError(
                "Render resolution is invalid. See log for details.", 
                description=(
                    "Wrong render resolution setting. "
                    "Please use repair button to fix it.\n"
                    "If current renderer is V-Ray, "
                    "make sure vraySettings node has been created"
                )
            )
```